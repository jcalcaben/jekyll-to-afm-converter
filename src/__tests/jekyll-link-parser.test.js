import { unified } from 'unified';
import markdown from "remark-parse";

import remover from "../jekyll-link-parser";

test("parses jekyll link tags", () => {
  const tree = unified().use(markdown).parse(`This is a [link][]

[link]: {% link www.google.com %}
`);

  unified()
    .use(remover)
    .run(tree, (err, result) => {
      expect(result).toMatchSnapshot();
    });
});

test.todo("parses jekyll link tag in reference links");
