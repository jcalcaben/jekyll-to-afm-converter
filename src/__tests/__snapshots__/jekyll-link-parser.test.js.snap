// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parses jekyll link tags 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "position": Object {
            "end": Object {
              "column": 19,
              "line": 1,
              "offset": 18,
            },
            "start": Object {
              "column": 1,
              "line": 1,
              "offset": 0,
            },
          },
          "type": "text",
          "value": "This is a [link][]",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 19,
          "line": 1,
          "offset": 18,
        },
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "paragraph",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "[link]: {% link www.",
        },
        Object {
          "type": "text",
          "value": "yahoo",
        },
        Object {
          "type": "text",
          "value": ".com %}",
        },
      ],
      "position": Object {
        "end": Object {
          "column": 34,
          "line": 3,
          "offset": 53,
        },
        "start": Object {
          "column": 1,
          "line": 3,
          "offset": 20,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 1,
      "line": 4,
      "offset": 54,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
